services/src/main/java/org/exoplatform/addons/gamification/upgrade/DomainMigrationService.java

```diff
@@ -0,0 +1,176 @@
+package org.exoplatform.addons.gamification.upgrade;
+
+
+import org.exoplatform.addons.gamification.entities.domain.effective.GamificationActionsHistory;
+import org.exoplatform.addons.gamification.service.configuration.BadgeService;
+import org.exoplatform.addons.gamification.service.configuration.DomainService;
+import org.exoplatform.addons.gamification.service.configuration.RuleService;
+import org.exoplatform.addons.gamification.service.dto.configuration.BadgeDTO;
+import org.exoplatform.addons.gamification.service.dto.configuration.DomainDTO;
+import org.exoplatform.addons.gamification.service.dto.configuration.RuleDTO;
+import org.exoplatform.addons.gamification.service.mapper.DomainMapper;
+import org.exoplatform.addons.gamification.storage.dao.GamificationHistoryDAO;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.picocontainer.Startable;
+
+import java.util.List;
+
+public class DomainMigrationService implements Startable {
+
+    private static final Log LOG = ExoLogger.getLogger(DomainMigrationService.class);
+
+
+
+    protected RuleService ruleService;
+    protected DomainService domainService;
+    protected BadgeService badgeService;
+    protected GamificationHistoryDAO gamificationHistoryDAO;
+    protected DomainMapper domainMapper;
+
+    public DomainMigrationService(RuleService ruleService, DomainService domainService, BadgeService badgeService, GamificationHistoryDAO gamificationHistoryDAO, DomainMapper domainMapper) {
+
+        this.ruleService = ruleService;
+        this.domainService = domainService;
+        this.badgeService = badgeService;
+        this.gamificationHistoryDAO = gamificationHistoryDAO;
+        this.domainMapper = domainMapper;
+    }
+
+    @Override
+    public void start() {
```

Comment:
Is there a reason to not use an upgrade plugin and start the migration at each PLF restart ?

Reply:
We don't use an upgrade plugin since it's nor depending on the product version, that is why we are using a startable service that check if the upgrade is done or not then do the upgrade, the way how we check will be updated in the coming change.

Reply:
ok good point I missed that